crates/typst-syntax/src/node.rs

```diff
@@ -381,6 +398,11 @@ struct InnerNode {
     descendants: usize,
     /// Whether this node or any of its children are erroneous.
     erroneous: bool,
+    /// The (capped) amount of newlines in this node's descendants.
+    /// This is solely used to tell whether this node contains 0, 1, 2 or more
+    /// newlines. As such, this number is capped at 2, even though there may be
+    /// more newlines inside this node.
+    capped_newlines: u8,
```

Comment:
do we really need a field for this? is it too slow otherwise?

Reply:
Hmmm. I didn't really measure it, but it seemed like a waste to compute this twice. However, right now, we're _always_ computing this, even when there are no warnings, which is evidently worse. Something like an `OnceCell` for each node could be ideal, but I'll probably just remove this field for now.